python/lsst/alert/packet/updateSchema.py

```diff
@@ -0,0 +1,218 @@
+                             "type": ["null", str(column["datatype"])],
+                             "doc": "", "default": None}
+                    field_dictionary_array.append(field)
+
+    return field_dictionary_array
+
+
+def create_schema(name, field_dictionary_array, version):
+    """ Create a schema using the field dictionary. fastavro will automatically
+    take the name and namespace and put them as one, so the name should just be
+    the schema name and the namespace needs to be created separately. The
+    fastavro keys also need to be removed from the schema.
+
+        Parameters
+    ----------
+    name: `string`
+        The name of the schema as a string. E.G. diaSource.
+
+    field_dictionary_array: 'np.array'
+        An array containing dictionary entries for the individual fields.
+
+    version: 'string'
+        The version number of the schema
+    """
+    name = name[0:2].lower() + name[2:]
+    schema = fastavro.parse_schema({
+        "name": name,
+        "type": "record",
+        "fields": field_dictionary_array
+    })
+
+    schema['namespace'] = 'lsst.v' + version
+    fastavro_keys = list(schema.keys())
+    for key in fastavro_keys:
+        if '__' in key and '__len__' not in key:
+            schema.pop(key)
+
+    schema = dict(sorted(schema.items(), reverse=True))
+
+    return schema
+
+
+def update_schema(apdb_filepath, update_version=None):
+    """Compare an avro schemas  docstrings with the apdb.yaml file.
+
+    If there are no docstrings, add the field. If the docstrings do
+    not match, update the schema docstrings to the apdb docstrings.
+
+    Once it is updated, write out the new schema.
+
+    Parameters
+    ----------
+    apdb_filepath: `string`
+        Input string for the apdb.yaml file where the docstrings
+        will be compared.
+
+    update_version: 'string'
+        If a string is included, update schema to provided version.
+        Example: "5.1"
+
+    """
+
+    registry = SchemaRegistry.from_filesystem()
+
+    for version in registry.known_versions:
+        schema_root = get_schema_root()
+        path = os.path.join(schema_root, *version.split("."))
+
+        with open(apdb_filepath, 'r') as file:
+            apdb = yaml.safe_load(file)
+
+        if update_version:
+            version_name = update_version.split(".")[0] + "_" + update_version.split(".")[1]
+        else:
+            version_name = version.split(".")[0] + "_" + version.split(".")[1]
+
+        # The first 4 columns in the apdb are the ones we use for alerts
+        for x in range(0, 4):
+
+            name = apdb['tables'][x]['name']
+            field_dictionary = populate_fields(apdb['tables'][x])
+            schema = create_schema(name, field_dictionary, version_name)
+
+            write_schema(schema, version_name, path)
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(description='Update the schema docstrings so that they'
```

Comment:
again, I'd update the docstring here: rather than updating docstrings we're generating a new schema set from apdb.yaml.

Reply:
Changed the docstring to reflect that we are generating new schemas.